### ISSUE TITLE
Using `DataProviders` with abstract test method parameters results in `NullPointerExceptions`

### ISSUE THREAD

**Issue by csvtuda** [Qytera-Gmbh/QTAF#47]
# Description

**Edit**: Applies to interfaces/abstract classes in general, see https://github.com/Qytera-Gmbh/QTAF/issues/47#issuecomment-1446021835

When using a `DataProvider` for test cases that require a `java.nio.file.Path` as input parameter, QTAF fails to properly initialize the log collection:

```java
rx.exceptions.OnErrorNotImplementedException: Cannot invoke "de.qytera.qtaf.core.log.model.collection.TestScenarioLogCollection.addLogMessage(de.qytera.qtaf.core.log.model.message.LogMessage)" because the return value of "de.qytera.qtaf.core.context.IQtafTestContext.getLogCollection()" is null

	at rx.internal.util.InternalObservableUtils$ErrorNotImplementedAction.call(InternalObservableUtils.java:386)
	at rx.internal.util.InternalObservableUtils$ErrorNotImplementedAction.call(InternalObservableUtils.java:383)
	at rx.internal.util.ActionSubscriber.onError(ActionSubscriber.java:44)
	at rx.observers.SafeSubscriber._onError(SafeSubscriber.java:153)
	at rx.observers.SafeSubscriber.onError(SafeSubscriber.java:115)
	at rx.exceptions.Exceptions.throwOrReport(Exceptions.java:212)
	at rx.observers.SafeSubscriber.onNext(SafeSubscriber.java:139)
	at rx.subjects.PublishSubject$PublishSubjectProducer.onNext(PublishSubject.java:304)
	at rx.subjects.PublishSubject$PublishSubjectState.onNext(PublishSubject.java:219)
	at rx.subjects.PublishSubject.onNext(PublishSubject.java:72)
	at de.qytera.qtaf.core.guice.method_interceptor.QtafStepTrackerInterceptor.invoke(QtafStepTrackerInterceptor.java:40)
	at com.project.TestContext$DataProviderTest.testProvider(TestContext.java:66)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
	at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:677)
	at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:221)
	at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
	at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:969)
	at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:194)
	at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
	at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at org.testng.TestRunner.privateRun(TestRunner.java:829)
	at org.testng.TestRunner.run(TestRunner.java:602)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:437)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:431)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:391)
	at org.testng.SuiteRunner.run(SuiteRunner.java:330)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1256)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1176)
	at org.testng.TestNG.runSuites(TestNG.java:1099)
	at org.testng.TestNG.run(TestNG.java:1067)
	at com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:66)
	at com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:109)
Caused by: java.lang.NullPointerException: Cannot invoke "de.qytera.qtaf.core.log.model.collection.TestScenarioLogCollection.addLogMessage(de.qytera.qtaf.core.log.model.message.LogMessage)" because the return value of "de.qytera.qtaf.core.context.IQtafTestContext.getLogCollection()" is null
	at de.qytera.qtaf.core.event_subscriber.step.StepLoggerSubscriber.onBeforeStepExecution(StepLoggerSubscriber.java:96)
	at rx.internal.util.ActionSubscriber.onNext(ActionSubscriber.java:39)
	at rx.observers.SafeSubscriber.onNext(SafeSubscriber.java:134)
	... 32 more
```

# Steps to reproduce
Minimal working example:
```java

package com.project;

import de.qytera.qtaf.core.config.annotations.TestFeature;
import de.qytera.qtaf.core.guice.annotations.Step;
import de.qytera.qtaf.testng.context.QtafTestNGContext;
import org.testng.Assert;
import org.testng.annotations.DataProvider;
import org.testng.annotations.Test;

import javax.inject.Inject;
import javax.inject.Singleton;
import java.nio.file.Path;
import java.nio.file.Paths;

public class TestContext extends QtafTestNGContext {

    @Inject
    protected static DataProviderTest.SomePage somePage;

    @Singleton
    @TestFeature(name = "123", description = "456")
    public class DataProviderTest extends TestContext {

        @DataProvider(name = "provider")
        public Object[][] providePaths() {
            return new Object[][]{
                    {
                            Paths.get(
                                    "parent",
                                    "child"
                            )
                    }
            };
        }

        @Test(testName = "xyz", dataProvider = "provider")
        public void testProvider(Path path) {
            Assert.assertTrue(somePage.isOk());
        }

        @Singleton
        public static class SomePage extends QtafTestNGContext {

            public SomePage() {

            }

            @Step(name = "is ok")
            public boolean isOk() {
                return true;
            }

        }

    }
}
```
Executing the `testProvider` test case will result in the above error message. Moving all inner classes into individual files does not change the outcome. Adding `@Singleton` to the `TestContext` does not help either.

Switching to `String` instead of `Path` makes the test pass.

**Comment by csvtuda** [comment]
The problem seems to occur whenever interfaces are used as test case parameters.
```java
    private interface DebugInterface {
        void doSomething();
    }

    private static class DebugClass implements DebugInterface {
        @Override
        public void doSomething() {
            System.out.println("Hi");
        }
    }

    @DataProvider(name = "provider")
    public Object[][] providePaths() {
        return new Object[][]{
                {
                        new DebugClass()
                }
        };
    }

    @Test(testName = "xyz", dataProvider = "provider")
    public void testProvider(DebugInterface path) {
        Assert.assertTrue(somePage.isOk());
    }
```

**Comment by csvtuda** [comment]
Fixed in #64.
